Source/WebCore/style/Styleable.cpp

```diff
@@ -972,13 +972,14 @@ void Styleable::queryContainerDidChange() const
     auto* animations = this->animations();
     if (!animations)
         return;
-    for (auto animation : *animations) {
-        auto* cssAnimation = dynamicDowncast<CSSAnimation>(animation.get());
-        if (!cssAnimation)
-            continue;
-        auto* keyframeEffect = dynamicDowncast<KeyframeEffect>(cssAnimation->effect());
-        if (keyframeEffect && keyframeEffect->blendingKeyframes().usesContainerUnits())
-            cssAnimation->keyframesRuleDidChange();
+    for (auto& animation : *animations) {
+        RefPtr keyframeEffect = dynamicDowncast<KeyframeEffect>(animation->effect());
```

Comment:
To be sure, `effect()` here can return a variety of types? (We always seem to `dynamicDowncast` to `KeyframeEffect`.)